### DISCUSSION TITLE
2023-06-23 WhoBPyT Sprint

### DISCUSSION THREAD

**Discussion by Andrew-Clappison** [GriffithsLab/whobpyt#92] [Meeting Minutes]
**Possible Tasks for the WhoBPyT Sprint. List is in no particular order and items are of varying difficulty and time commitment.**  
1.	Convert the synthetic data generated in the “Fitting Mean S_E example” to functions located in the “data/” folder that can be imported, then use these function(s) to simplify the example code.  - John
2.	Look into why some parameters currently don’t seem to be changing in Wong_Wang and JR model examples. - Zheng
3.	Create functions or a class to produce FC plots from a Recording object. Use these functions within the sphinx examples. -Parsa, Kevin
4.	Create functions or a class to produce PSD plots from a Recording object. Use these functions within the sphinx examples. -Parsa, Kevin
5.	Create functions or a class to produce TS plots from a Recording object. Use these functions within the sphinx examples. -Parsa, Kevin
6.	More advanced visual functionality. -Davide
7.	Updates to the Sphinx “Overview” page. -Sorenza
8.	Documentation of the objective function code (inline & readme). - Taha
9.	Functionality for objective function to take in empirical FC as input instead of just empirical time series.- Andrew
10.	Functionality for objective function to take in empirical PSD as input instead of just empirical time series. - Andrew
11.	Description of what the Parcellated Lead Field matrices are and what software/functions to create them. - Davide
12.	BOLD readme file info. - Shrey
13.	Remove some features/parameters to simplify RWW PyTorch code and make it easier to follow OOP style and have clean code, particularly for the ML analytics. Example: remove mean and variance of simulated time series state value parameters. - Zheng
14.	Convert RWW code to use log value so that the parameter returned by par() is actual value used (instead of min + ReLU(stored_value) being used).
15.	Convert Jansen-Rit to use log value so that the parameter returned by par() is actual value used (instead of min + ReLU(stored_value) being used).
16.	Simplify and/or document and explain the sampling feature of RWW to BOLD. - Zheng
17.	RWW Validation Code & Add to RWW example - Sorenza
18.	JR Validation Code & Add to JR example - Sorenza
19.	JR Model README - Shrey
20.	Plasticity Idea
21.	Someone to go through and make a list of recommended filename/variable name changes to bring consistency across the package. Of note, consider renaming RWW to RWWEI or something else. 
22.	Reproduce PyTepFit Paper Results in Sphinx example, where the code uses WhoBPyT but has static images instead of running the code on every pull request. - Davide
23.	Functions to import neuroimaging data from public online sources. - John
24.	Cleanup of WhoBPyT examples appearance in general. - John
25.	Updating the Linear Model to current approach.
26.	Inline documentation of datatypes and the README file for the folder. - Andrew
27.	Generate some automatic testing. 
28.	Manually test the current code as directed by @JohnWangDataAnalyst 
29.  Add type definitions to methods and classes - Parsa, Kevin

**Comment by JohnGriffiths** [comment]
# JG Notes from this meeting:


## Small to-dos

- Put installation instructions in readme  
- Do google colab usage instructions  

## Use decorators more?

Can be a lighter alternative to `.value()` calls in many cases


## Nomenclature

FYI, some names of classes etc will need to be changed eventually. But that isn't top priority right now so we'll punt it to a later time.

For example: 
- Some class names (eg 'RNN' everywhere, not needed)
- All the `torch.`s everywhere. Not needed, impairs readability. Just import the torch method. 
- Model nomenclature. E.g. Jansen-Rit, RWWExcInh. Various considerations as to what approach to take here.
- Use of terms "validate" , 'test',  'evaluate'


## Annotations

Lots of locations in the code where comments should be added explaining what is being done

A good code base has more lines of comments than it does code!


## Add type definitions ?

In specification of classes + methods

**Comment by sbastiaens** [comment]
Code written for jansen_rit_validate.py on local computer: still need to test it, having trouble importing modules.

**Comment by Andrew-Clappison** [comment]
**List of remaining items that may still need to be added as github issues:**

Functionality for objective function to take in empirical PSD as input instead of just empirical time series.

Convert RWW code to use log value so that the parameter returned by par() is actual value used (instead of min + ReLU(stored_value) being used).

Convert Jansen-Rit to use log value so that the parameter returned by par() is actual value used (instead of min + ReLU(stored_value) being used).

Remove Sampling feature from RWW to BOLD

RWW Validation Code & Add to RWW example 

Plasticity Idea

Someone to go through and make a list of recommended filename/variable name changes to bring consistency across the package. Of note, consider renaming RWW to RWWEI or something else.

Updating the Linear Model to current approach.
